internal/registry/registry.go

```diff
@@ -129,7 +135,7 @@ func (r *registry) getPullOptions(ctx context.Context, image string, po *ootov1a
 		crane.WithContext(ctx),
 	}
 
-	if po != nil {
+	if *po != (ootov1alpha1.PullOptions{}) {
```

Comment:
why not just compare to nil?

Reply:
Because `PullOption` isn't a pointer (in its definition context), which means that if it wasn't set by the user, its default value will be an empty struct, and when passed by reference to `getPullOptions` it will get the value of `&ootov1alpha1.PullOptions{}` and not `nil`.

I do see though that we are sometimes calling this function with the explicit `nil` arg so I will change the condition to `if po != nil && *po != (ootov1alpha1.PullOptions{})`.

Reply:
What i mean is that even default values of PullOptions are still valid values. So we can just check the pointer not nil, and proceed to update according to the default values. If in the future default values will have to be used for some setting, comparing to ootov1alpha1.PullOptions{}) will cause buggy flow.

Reply:
I see what you mean. Fixed it as well. Thanks.